airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -45,41 +50,52 @@
 import io.airbyte.protocol.models.ConnectorSpecification;
 import io.airbyte.protocol.models.Field;
 import io.airbyte.protocol.models.Field.JsonSchemaPrimitive;
+import io.airbyte.protocol.models.SyncMode;
 import java.io.IOException;
+import java.sql.Connection;
+import java.sql.JDBCType;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 import java.time.Instant;
-import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import org.jooq.DSLContext;
-import org.jooq.DataType;
-import org.jooq.JSONFormat;
-import org.jooq.JSONFormat.RecordFormat;
-import org.jooq.SQLDialect;
-import org.jooq.Schema;
-import org.jooq.Table;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class AbstractJdbcSource implements Source {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(AbstractJdbcSource.class);
 
-  private static final JSONFormat DB_JSON_FORMAT = new JSONFormat().recordFormat(RecordFormat.OBJECT);
-
   private final String driverClass;
-  private final SQLDialect dialect;
 
-  public AbstractJdbcSource(final String driverClass, final SQLDialect dialect) {
+  public AbstractJdbcSource(final String driverClass) {
     this.driverClass = driverClass;
-    this.dialect = dialect;
   }
 
+  /**
+   * Map a database implementation-specific configuration to json object that adheres to the
+   * AbstractJdbcSource config spec. See resources/spec.json.
+   *
+   * @param config database implementation-specific configuration.
+   * @return jdbc spec.
+   */
   public abstract JsonNode toJdbcConfig(JsonNode config);
 
+  /**
+   * Set of schemas that are internal to the database (e.g. system schemas) and should not be included
+   * in the catalog.
+   *
+   * @return set of schemas to be ignored.
+   */
+  public abstract Set<String> getExcludedInternalSchemas();
```

Comment:
Why not expose them and let the user decide if they want them or not? 
With the current implementation you're making a choice for someone (I am sure there is at least one person who will want these schemas). You also need to maintain a hardcoded list for every single implementation.

Reply:
discussed offline. some databases have a _huge_ number of internal tables (>400, ahem MySQL). when we run into the described problem we will make it configurable to allow system tables, but for now the 99th percentile use case should be that they are not needed.